components/bt/host/bluedroid/btc/profile/std/sdp/btc_sdp.c

```diff
@@ -274,17 +275,25 @@ static int add_raw_sdp(const bluetooth_sdp_record* rec)
 
     if (rec->hdr.bt_uuid.len == 16) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid128[2], sizeof(service));
+        UINT8_TO_BE_STREAM (p_temp, (UUID_DESC_TYPE << 3) | SIZE_SIXTEEN_BYTES);
+        ARRAY_TO_BE_STREAM (p_temp, rec->hdr.bt_uuid.uuid.uuid128, LEN_UUID_128);
     } else if (rec->hdr.bt_uuid.len == 2) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid16, sizeof(service));
+        UINT8_TO_BE_STREAM (p_temp, (UUID_DESC_TYPE << 3) | SIZE_TWO_BYTES);
+        UINT16_TO_BE_STREAM (p_temp, rec->hdr.bt_uuid.uuid.uuid16);
     } else if (rec->hdr.bt_uuid.len == 4) {
         memcpy(&service, &rec->hdr.bt_uuid.uuid.uuid16, sizeof(service));
```

Comment:
I want to convert 32 bit or 128 bit uuid to 16 bit uuid.

Reply:
Hi!
What is the purpose of converting them to 16-bit UUID?

I dug down and realized that we must patch [bta_sys_add_uuid(UINT16 uuid16)](https://github.com/espressif/esp-idf/blob/a4afa44435ef4488d018399e1de50ad2ee964be8/components/bt/host/bluedroid/bta/sys/bta_sys_conn.c#L528) and other functions in the chain to support 32 and 128 bit UUIDs as well.
I will do that to fix the issue https://github.com/espressif/esp-idf/issues/11529 , but it exceeds this pull request.

Reply:
Maybe we don't need to modify `bta_sys_add_uuid`. Just need to convert them to 16 bit uuid.

Reply:
That is not the topic of this PR, but I want to explain myself:

Bluedroid in some places has the ability to use 128-bit UUID.
For example, in the [bta_dm_set_eir()](https://github.com/espressif/esp-idf/blob/3a45d4e949a174e8829a2e4c86c421b030ceac5a/components/bt/host/bluedroid/bta/dm/bta_dm_act.c#L4212) code checks for the 128-bit UUIDs.
But the problem is that these 128 and 32-bit UUIDs are nowhere saved in the [bta_dm_cb.custom_uuid[] ](https://github.com/espressif/esp-idf/blob/3a45d4e949a174e8829a2e4c86c421b030ceac5a/components/bt/host/bluedroid/bta/dm/bta_dm_act.c#L4222)array. Only 16-bit values are stored.
That's why I want to modify `bta_sys_add_uuid` one day.

Reply:
you are right. In order to support 128 and 32-bit UUIDs need to modify `bta_sys_add_uuid`.